src/Components/Products/Products.js

```diff
@@ -4,27 +4,27 @@ import "./style.css";
 export default class Products extends Component {
   constructor(props) {
     super(props);
-    this.state = { selectIndex: null };
+    this.state = { select:null };
   }
   handleAdd = () => {
     this.props.addCart(this.props.index);
   };
   render() {
+    const {name,price,storage,image}=this.props.products
     return (
       <>
-        <div className="parent-div">
+        <div className="parent-div" key={this.props.index}>
```

Comment:
Is index a suitable key here?

Reply:
You've sent both index and key as a prop to this component, yet instead of using the key prop here, you've used index....